amazonsqs/amazonsqs_connector.bal

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/crypto;
+import ballerina/encoding;
+import ballerina/http;
+import ballerina/internal;
+import ballerina/io;
+import ballerina/system;
+import ballerina/time;
+import wso2/amazoncommons;
+import ballerina/log;
+
+# Object to initialize the connection with Amazon SQS.
+#
+# + accessKey - The Amazon API access key
+# + secretKey - The Amazon API secret key
+# + region - The Amazon API Region
+# + acctNum - The account number of the SQS service
+public type Client client object {
+
+    http:Client clientEp;
+    string accessKey;
+    string secretKey;
+    string region;
+    string acctNum;
+    string host;
+
+    public function __init(Configuration config) {
+        self.accessKey = config.accessKey;
+        self.secretKey = config.secretKey;
+        self.acctNum = config.accountNumber;
+        self.region = config.region;
+        self.host = SQS_SERVICE_NAME + "." + self.region + "." + AMAZON_HOST;
+        self.clientEp = new("https://" + self.host);
```

Comment:
Are we validating amazon configuration at the init level? Can we check whether the credentials are empty or not and check whether is there any API provided by amazon to valid the credentials. If the credentials are invalid, it should fail at the init level.

Reply:
No need as there is no proper validation specification was published by Amazon

Reply:
Still we can check the emptiness of the  credentials